src/shared/cpu-set-util.c

```diff
@@ -290,3 +307,55 @@ int cpu_set_from_dbus(const uint8_t *bits, size_t size, CPUSet *set) {
         *set = TAKE_STRUCT(s);
         return 0;
 }
+
+static int get_max_cpu(void) {
+        _cleanup_closedir_ DIR *d = NULL;
+        int r, max_cpu = 0;
+
+        d = opendir("/sys/devices/system/cpu");
+        if (!d)
+                return -errno;
+
+        FOREACH_DIRENT(de, d, break) {
+                int cpu;
+                const char *c;
+
+                if (de->d_type != DT_DIR)
+                        continue;
+
+                c = startswith(de->d_name, "cpu");
+                if (!c)
+                        continue;
+
+                r = safe_atoi(c, &cpu);
+                if (r < 0)
+                        continue;
+
+                if (cpu > max_cpu)
+                        max_cpu = cpu;
+        }
+
+        return max_cpu;
+}
+
+int cpu_mask_add_all(CPUSet *mask) {
```

Comment:
uh, so we have cpu_set_add_all(), and now cpu_mask_add_all() which sound very similar, but do something quite different...

Reply:
The former is used to copy from NUMA mask.